float64 x_c
float64 theta_c
float64 f_c

float64 x_e
float64 y_e
float64 theta_e

float64 x_c_hat
float64 theta_c_hat
float64 f_c_hat
